### ISSUE TITLE
Add more Math operation support System.Numerics.Vectors

### ISSUE THREAD

**Issue by dangi12012** [dotnet/corefx#22440]
Support for some SIMD instructions are missing in Numerics.Vectors

Vector.RootN //Only SquareRoot is implemented
Vector.Exp //Every element of the result is exp(a,b) where a,b are input vectors
Vector.LShift //Shift elements in a by elements in b
Vector.RShift //Like above would simulate a = b >> c for all elements

It wondsers me because Or and XOr is there but not shift. 
Code where Vectors would be perfect but are not supported yet:
```
while (pbuf < pend)
{
    uint tx = x ^ (x << 11);
    uint ty = y ^ (y << 11);
    uint tz = z ^ (z << 11);
    uint tw = w ^ (w << 11);
    *(pbuf++) = x = w ^ (w >> 19) ^ (tx ^ (tx >> 8));
    *(pbuf++) = y = x ^ (x >> 19) ^ (ty ^ (ty >> 8));
    *(pbuf++) = z = y ^ (y >> 19) ^ (tz ^ (tz >> 8));
    *(pbuf++) = w = z ^ (z >> 19) ^ (tw ^ (tw >> 8));
}
```

**Comment by tannergooding** [comment]
For `Vector.RootN`, There is no hardware acceleration (no SIMD instruction for this). Additionally, even `System.Math` and `System.MathF` don't currently expose a `RootN` function so there would not be a 'simple' way of creating a pure software implementation today.

For `Vector.Exp`, There is also no hardware acceleration but writing an extension method would be trivial. I am also not sure if this is a scenario that warrants a new API on the `Vector` structs.

For `LShift` and `RShift`, there is an existing proposal: https://github.com/dotnet/corefx/issues/16835

**Comment by Drawaes** [comment]
I would like Exp it's very important for brownian motion calculations used in monte carlos. There is a whole rang of hardware acceleration for these.

https://software.intel.com/en-us/node/695443

**Comment by Drawaes** [comment]
I take it back while intel calls them "intrinsics" they are actually just functions with multiple AVX calls.... pity

**Comment by tannergooding** [comment]
@Drawaes, yeah. There is no actual hardware instruction for these, so implementing it becomes non-trivial.

If we had some form of `raw intrinsic` support in CoreFX (https://github.com/dotnet/coreclr/issues/6906#issuecomment-307164495), then this would at least be somewhat 'easier' to implement (reimplement `MathF.Exp` using the intrinsic APIs and then implement a `Vector.Exp` form of that algorithm).

For now, having an extension method and calling `MathF.Exp` four times is the best we can do.

**Comment by dangi12012** [comment]
If you already implement extension Methods think about: Bitshift, Modulus and Exp. Maybe even Sin, Cos. Then you already have them implemented if they ever are accelerated by hardware!

**Comment by tannergooding** [comment]
Would become easier to implement with: https://github.com/dotnet/designs/issues/13

**Comment by fiigii** [comment]
Intel hardware intrinsic API proposal has been opened at #22940

**Comment by mjmckp** [comment]
I have implementations of `Log` and `Exp` for `System.Numerics.Vector` (see https://github.com/mjmckp/VectorMathFuns and https://github.com/dotnet/corefx/issues/15957), however these require vectorised versions of `ceil` and `floor` (https://github.com/dotnet/corefx/issues/16833), and left/right bit shifting (https://github.com/dotnet/corefx/issues/16835).  Unfortunately, the proposal for bit shifting has stalled.

**Comment by 4creators** [comment]
> the proposal for bit shifting has stalled

Quite the opposite see: https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Intrinsics.X86/ref/System.Runtime.Intrinsics.X86.cs

Functionality is actively worked on in CoreCLR repo. The plan is to support expanded Numerics.Vector functionality with Intrinsics.

**Comment by mjmckp** [comment]
@4creators Awesome, thanks!

**Comment by dangi12012** [comment]
Oh yes. I see.  hope Span<T> support will be enabled for vectors soon. To define your own operators instead of this:
`
        /// Floor
        public static Vector<float> _mm256_floor_ps(Vector<float> x)
        {
            var rslt = new float[Vector<float>.Count];
            for (int i = 0; i < rslt.Length; ++i)
                rslt[i] = (float) Math.Floor(x[i]);
            return new Vector<float>(rslt);
        }`
